### PR TITLE
Added capacity to slice creation, when capacity is known

### PR DESCRIPTION
**Pull request by vivekv96**
Simple optimisation to reduce memory allocations and copying(while appending).

### CODE REVIEW

**Comment by vivekv96** [comment]
@googlebot I signed it!

File: internal/report/report.go [hunk 1]
```diff
@@ -386,7 +386,7 @@ func PrintAssembly(w io.Writer, rpt *Report, obj plugin.ObjTool, maxFuncs int) e
 	symNodes := nodesPerSymbol(g.Nodes, symbols)
 
 	// Sort for printing.
-	var syms []*objSymbol
+	syms := make([]*objSymbol, 0, len(symNodes))
```
Comment (nolanmar511) [inline internal/report/report.go]: I believe using creates an array of the specified size filled with empty values, then appending will add a non-empty value to the end (rather filling in an empty spot).
For example
```
arr := make([]int, 2)
arr = append(arr, 1)
```
produces the array [0, 0, 1].

Comment (vivekv96) [reply internal/report/report.go]: @nolanmar511, regarding the example that you have provided, yes that would produce [0, 0, 1].
But, the make() that I have used has length=0 and capacity=len(symNodes), so it's different from giving it a length of 2.

So, the equivalent example = 
```
arr := make([]int, 0, 5)
arr = append(arr, 1)
```

which produces the array [1]. Playground link: https://play.golang.org/p/nSUHnfqubiJ

**Comment by aalexand** [comment]
Also, does this change solve a specific performance problem? If not, the
code is fine as is, let's not optimize prematurely.

**Comment by vivekv96** [comment]
@aalexand No, I came across this while just glancing over the code, haven't faced any performance issues per se. 
While I agree with you that this is a premature optimisation of sorts, and the code works as is, this does reduce allocs/op and the other metrics.
If you believe that this is unnecessary, go ahead and close this PR.
Thanks. No worries.

**Pull request closed without merging**